openmdao/core/system.py

```diff
@@ -3631,7 +3631,19 @@ def list_outputs(self,
                 expl_outputs = list(expl_outputs.items())
 
         if implicit:
-            impl_outputs = {n: m for n, m in outputs.items() if n in states}
+            if residuals_tol:
+                for val in outputs.values():
+                    if not hasattr(val, "resids"):
+                        simple_warning("No residuals found, skipping residuals_tol")
+                        impl_outputs = {n: m for n, m in outputs.items() if n in states}
+                        break
+                    else:
+                        impl_outputs = {n: m for n, m in outputs.items()
+                                        if n in states and
+                                        (n in states and m['resids'] != residuals_tol)}
```

Comment:
per:
```
        residuals_tol : float, optional
            If set, limits the output of list_outputs to only variables where
            the norm of the resids array is greater than the given 'residuals_tol'.
```
I think this test should be ` np.linalg.norm(resids) > residuals_tol` 

See comparable logic here:
https://github.com/OpenMDAO/OpenMDAO/blob/master/openmdao/recorders/case.py#L587

Reply:
Whoops, fixed that now